src/force/CabanaPD_Force_PMB.hpp

```diff
@@ -0,0 +1,387 @@
+            f( i, 1 ) += fy_i;
+            f( i, 2 ) += fz_i;
+        };
+
+        Kokkos::RangePolicy<exec_space> policy( 0, n_local );
+        Cabana::neighbor_parallel_for(
+            policy, force_full, neigh_list, Cabana::FirstNeighborsTag(),
+            neigh_op_tag, "CabanaPD::ForcePMB::computeFull" );
+    }
+
+    template <class PosType, class WType, class ParticleType,
+              class NeighListType, class ParallelType>
+    double computeEnergyFull( WType& W, const PosType& x, const PosType& u,
+                              const ParticleType& particles,
+                              const NeighListType& neigh_list,
+                              const int n_local,
+                              ParallelType& neigh_op_tag ) const
+    {
+        auto c = _model.c;
+        const auto vol = particles.sliceVolume();
+
+        auto energy_full =
+            KOKKOS_LAMBDA( const int i, const int j, double& Phi )
+        {
+            // Get the bond distance, displacement, and stretch.
+            double xi, r, s;
+            getDistance( x, u, i, j, xi, r, s );
+
+            // 0.25 factor is due to 1/2 from outside the integral and 1/2 from
+            // the integrand (pairwise potential).
+            double w = 0.25 * c * s * s * xi * vol( j );
+            W( i ) += w;
+            Phi += w * vol( i );
+        };
+
+        double strain_energy = 0.0;
+        Kokkos::RangePolicy<exec_space> policy( 0, n_local );
+        Cabana::neighbor_parallel_reduce(
+            policy, energy_full, neigh_list, Cabana::FirstNeighborsTag(),
+            neigh_op_tag, strain_energy,
+            "CabanaPD::ForcePMB::computeEnergyFull" );
+
+        return strain_energy;
+    }
+};
+
+template <class ExecutionSpace>
+class Force<ExecutionSpace, ForceModel<PMB, Fracture>>
+    : public Force<ExecutionSpace, ForceModel<PMB, Elastic>>
+{
+  protected:
+    using base_type = Force<ExecutionSpace, ForceModel<PMB, Elastic>>;
+    using base_type::_half_neigh;
+    ForceModel<PMB, Fracture> _model;
+
+  public:
+    using exec_space = ExecutionSpace;
+
+    Force( const bool half_neigh, const ForceModel<PMB, Fracture> model )
+        : base_type( half_neigh, model )
+        , _model( model )
+    {
+    }
+
+    template <class ForceType, class PosType, class ParticleType,
+              class NeighListType, class MuView, class ParallelType>
+    void computeForceFull( ForceType& f, const PosType& x, const PosType& u,
+                           const ParticleType& particles,
+                           const NeighListType& neigh_list, MuView& mu,
+                           const int n_local, ParallelType& ) const
+    {
+        auto c = _model.c;
+        auto break_coeff = _model.bond_break_coeff;
+        const auto vol = particles.sliceVolume();
+        const auto nofail = particles.sliceNoFail();
+
+        auto force_full = KOKKOS_LAMBDA( const int i )
+        {
+            std::size_t num_neighbors =
```

Comment:
@streeve Why we do not have this in computeForceFull in Line 101?

Reply:
The fracture and elastic are different - we need different information for using the `mu` array